fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -293,22 +292,29 @@ public boolean killTask(String taskName, boolean clearPending) {
         return taskRunManager.killTaskRun(task.getId());
     }
 
-    public Constants.TaskRunState executeTaskSync(String taskName) throws ExecutionException, InterruptedException {
-        return executeTaskSync(taskName, new ExecuteOption());
+    public String executeTaskSync(Task task) {
+        return executeTaskSync(task, new ExecuteOption());
     }
 
-    public Constants.TaskRunState executeTaskSync(String taskName, ExecuteOption option)
-            throws ExecutionException, InterruptedException {
-        Task task = nameToTaskMap.get(taskName);
-        if (task == null) {
-            throw new DmlException("execute task:" + taskName + " failed");
+    public String executeTaskSync(Task task, ExecuteOption option) {
+        if (!tryTaskLock()) {
```

Comment:
You get the task lock synchronously here, then if 2 people submit different. Can one person influence another?
This person didn't complete the execution, and the other person got stuck?

Reply:
Sync mode should execute the fresh task linearly?

Reply:
here should not hold the lock during task execution. For the same task, task run manager will make sure that the task run is scheduled one by one.